library/src/main/java/com/datatorrent/lib/io/fs/FileSplitterInput.java

```diff
@@ -375,11 +375,14 @@ public void run()
             lastScannedInfo = null;
             numDiscoveredPerIteration = 0;
             for (String afile : files) {
-              String filePath = new File(afile).getAbsolutePath();
+              Path filePath = new Path(afile);
               LOG.debug("Scan started for input {}", filePath);
-              Map<String, Long> lastModifiedTimesForInputDir;
-              lastModifiedTimesForInputDir = referenceTimes.get(filePath);
-              scan(new Path(afile), null, lastModifiedTimesForInputDir);
+              Map<String, Long> lastModifiedTimesForInputDir = null;
+              if (fs.exists(filePath)) {
+                FileStatus fileStatus = fs.getFileStatus(filePath);
+                lastModifiedTimesForInputDir = referenceTimes.get(fileStatus.getPath().toUri().getPath());
```

Comment:
No. In createScannedFileInfo(), the directory/file path specified as absolute path. But, In case of local file system, files might be relative path.